src/invertedindex.js

```diff
@@ -0,0 +1,222 @@
+
+const hasProperty = Object.prototype.hasOwnProperty;
+let instance = null;
+
+class InvertedIndex {
+  /**
+    * constructor method
+  */
+  constructor() {
+    if (!instance) {
```

Comment:
The constructor does not take any parameters. This block will always be hit. I thnk it makes sense to take it out of the conditional.